ui/src/publication/log/publication-log.tsx

```diff
@@ -159,10 +125,84 @@ function getSearchableItemName(
     }
 }
 
+const isValidPublicationLogSearchRange = (
+    specificItem: SearchItemValue<SearchablePublicationLogItem> | undefined,
+    start: Date | undefined,
+    end: Date | undefined,
+): boolean => {
+    return (
+        specificItem !== undefined || // go nuts with single-item searches, they're cheap
+        (start !== undefined && daysBetween(start, end ?? new Date()) < MAX_SEARCH_DAYS)
+    );
+};
+
+function usePublicationLogSearch(
+    startDate: Date | undefined,
+    endDate: Date | undefined,
+    specificItem: SearchItemValue<SearchablePublicationLogItem> | undefined,
+    sortInfo: TableSorting<SortablePublicationTableProps>,
+): { pagedPublications: Page<PublicationTableItem> | undefined; isLoading: boolean } {
+    const [isLoading, setIsLoading] = React.useState(false);
+    const [pagedPublications, setPagedPublications] = React.useState<Page<PublicationTableItem>>();
+    const displayedSearchResultParams = React.useRef<
+        | {
+              startDate: Date | undefined;
+              endDate: Date | undefined;
+              specificItem: SearchItemValue<SearchablePublicationLogItem> | undefined;
+              sortInfo: TableSorting<SortablePublicationTableProps>;
+          }
+        | undefined
```

Comment:
Tämän tyypin voisi ehkä irrottaa omaksi määrityksekseen, niin ei tulisi näin pitkää reffin määritystä